oak_client/src/attestation.rs

```diff
@@ -91,13 +110,19 @@ impl RemoteAttestationVerifier {
         }
     }
 
-    fn check_tee_report(&self, extension: &str) -> anyhow::Result<()> {
+    /// Check that the [`extension`] contains [`RemoteAttestationVerifier::tee_measurement`] and
+    /// a SHA-256 digest of the [`public_key`].
+    fn check_tee_report(&self, extension: &str, public_key: &[u8]) -> anyhow::Result<()> {
         let tee_report = Report::from_string(extension).context("Couldn't parse TEE report")?;
-        if self.tee_measurement == tee_report.measurement {
-            Ok(())
-        } else {
-            Err(anyhow!("Incorrect TEE report"))
+        let public_key_hash = get_sha256(public_key);
```

Comment:
Should there be a check of the signature of the report too? Even if it is just a placeholder, that seems quite important.